### PR TITLE
feat: Adds ASL Opcode

### PR DESCRIPTION
**Pull request by rafaeldelboni**
### ASL (Arithmetic Shift Left)

Affects Flags: N Z C
ASL shifts all bits left one position. 0 is shifted into bit 0 and the original bit 7 is shifted into the Carry.

### CODE REVIEW

File: src/opcodes.c [hunk 1]
```diff
@@ -20,6 +20,26 @@ int hexToDecimalMode(unsigned char hex) {
...
 
+bool isAccumulatorAddressMode(unsigned int address) {
+  return address == 0;
+}
+
+void aslOpcode(unsigned int address, struct CPU *cpu, struct BUS *bus)
+{
+  unsigned char current = isAccumulatorAddressMode(address)
+    ? cpu->accumulator
+    : readBus(address, bus);
```
Comment (lucascebertin) [inline src/opcodes.c]: Do you know if we will need to use more of this code block?
If yes, what you think to extract this in a specific method?
Is it makes any sense?

File: src/opcodes.c [hunk 2]
```diff
@@ -20,6 +20,26 @@ int hexToDecimalMode(unsigned char hex) {
...
+  unsigned char result = current << 1;
+
+  cpu->status.carry = current & 0x80;
+  cpu->status.negative = result >> 7;
+  cpu->status.zero = result == 0x00;
+
+  return isAccumulatorAddressMode(address)
+    ? cpu->accumulator = result
+    : writeBus(address, result, bus);
```
Comment (lucascebertin) [inline src/opcodes.c]: Alright, we do have something similar but this time it's write not only reads...
Please consider whether my previous comment is appropiated for this one too or not.

File: src/opcodes.c (hunk 1 above)
Comment (rafaeldelboni) [reply src/opcodes.c]: :+1:

File: src/opcodes.c (hunk 2 above)
Comment (rafaeldelboni) [reply src/opcodes.c]: :+1:

**Review by lucascebertin — APPROVED** [review]
Great!
Let's merge this!

**Pull request merged**